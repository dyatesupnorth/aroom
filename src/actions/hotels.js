import axios from "axios";
axios.defaults.baseURL = 'http://localhost:3001/'
export const setHotels = hotels => ({
  type: "SET_HOTELS",
  hotels
});

export const startSetHotels = () => {
  return dispatch => {
    return axios.get("hotels").then(res => 
      dispatch(setHotels(res.data))
      );
  };
};
