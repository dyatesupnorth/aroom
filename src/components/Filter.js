import React, { Component } from "react";
import { connect } from "react-redux";
import {
  setTextFilter,
  setStarRatingFilter,
  setFacilitiesFilter
} from "../actions/filters";

export class Filter extends Component {
  
  onTextChange = e => {
    this.props.setTextFilter(e.target.value);
  };

  onStarRatingChange = e => {
    this.props.setStarRatingFilter(e.target.value);
  };

  onFacilitiesChange = (e, i) => {
    this.props.setFacilitiesFilter(e.target.value);
  };

  renderRadioButtons(maxStarRating) {
    // TODO: Pull max star rating dervied from max of all hotels
    // TODO: Convert to checkboxes, we want to be able to filter based on multiple selections
    let radioButtons = [];
    for (let i = 0; i < maxStarRating; i++) {
      radioButtons.push(
        <span key={i}>
          <input
            type="radio"
            value={i + 1}
            name="starRating"
            onChange={this.onStarRatingChange}
          />
          {i + 1}
        </span>
      );
    }
    return radioButtons;
  }

  renderCheckBoxes() {
    // TODO: Pull all available facilities directly from available hotels.
    return this.props.availableFacilities.map((facility, i) => (
      <label key={i}>
        {facility}
        <input
          name={facility}
          value={facility}
          type="checkbox"
          onChange={this.onFacilitiesChange}
        />
      </label>
    ));
  }

  render() {
    return (
      this.props.filters && (
        <div style={{ border: "1px solid tomato" }}>
          <h1>Filter</h1>
          <div className="form-group">
            <label>Name</label>
            <input
              className="form-control"
              name="text"
              type="text"
              value={this.props.filters.text}
              onChange={this.onTextChange}
            />
          </div>
          <div className="form-group">
            <label>Star Rating</label>
            {this.renderRadioButtons(5)}
          </div>
          <div className="form-group">
            <label>Facilities</label>
            {this.renderCheckBoxes()}
          </div>
        </div>
      )
    );
  }
}

const mapStateToProps = state => {
  console.log("â€‹state", state);
  return {
    filters: state.filters,
    // Trying to get too fancy!
    availableFacilities: state.hotels.reduce(
      (acc, hotel) => (
        [].push.apply(acc, hotel.facilities.filter(v => acc.indexOf(v) === -1)),
        acc
      ),
      []
    )
  };
};

const mapDispatchToProps = dispatch => {
  return {
    setTextFilter: text => dispatch(setTextFilter(text)),
    setStarRatingFilter: starRating =>
      dispatch(setStarRatingFilter(starRating)),
    setFacilitiesFilter: facility => dispatch(setFacilitiesFilter(facility))
  };
};
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Filter);
