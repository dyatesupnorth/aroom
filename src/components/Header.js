import React, { Component } from "react";

export class Header extends Component {
  render() {
    return (
      <div style={{ border: "1px solid tomato" }}>
        <h1>Header</h1>
      </div>
    );
  }
}

export default Header;
