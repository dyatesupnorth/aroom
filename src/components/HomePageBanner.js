import React, { Component } from "react";

export class HomePageBanner extends Component {
  render() {
    return (
      <div style={{ border: "1px solid tomato" }}>
        <h1>HomePageBanner</h1>
      </div>
    );
  }
}

export default HomePageBanner;
