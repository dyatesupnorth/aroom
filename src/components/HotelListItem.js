import React from "react";

export const HotelListItem = props => {
  return (
    <div style={{ border: "1px solid tomato" }}>
      <h2>{props.name}</h2>
      <p>Star rating: {props.starRating}</p>
      {!props.facilities || props.facilities.length === 0 ? (
        <p>No facilities</p>
      ) : (
        <p>
          Facilities:
          {props.facilities.map(facility => {
            return <span key={facility}>{facility}</span>;
          })}
        </p>
      )}
    </div>
  );
};

export default HotelListItem;
