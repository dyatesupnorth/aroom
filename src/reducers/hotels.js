const hotelsReducerDefaultState = [];
export default (state = hotelsReducerDefaultState, action) => {
  switch (action.type) {
    case "SET_EXPENSES":
      return action.hotels;
    default:
      return state;
  }
};
