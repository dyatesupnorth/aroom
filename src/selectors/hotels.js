export default (hotels, { text, starRating, facilities }) => {
  return hotels.filter(hotel => {
    const textMatch = hotel.name.toLowerCase().includes(text.toLowerCase());
    const starRatingMatch = starRating ? hotel.starRating >= starRating : true;
    const facilitiesMatch = facilities && facilities.length > 0 ? facilities.every(facility => hotel.facilities.includes(facility)) :true
    return textMatch && starRatingMatch && facilitiesMatch;
  });
};
