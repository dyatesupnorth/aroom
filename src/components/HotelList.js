import React from "react";
import { connect } from "react-redux";
import selectHotels from "../selectors/hotels";
import { HotelListItem } from "./HotelListItem";
import Card from "react-bootstrap/Card";

import ListGroup from "react-bootstrap/ListGroup";

export const HotelList = props => {
  return (
    <Card>
      <ListGroup variant="flush">
        {props.hotels.length === 0 ? (
          <p>No hotels</p>
        ) : (
          props.hotels.map(hotel => {
            return <HotelListItem key={hotel.name} {...hotel} />;
          })
        )}
      </ListGroup>
    </Card>
  );
};

const mapStateToProps = state => {
  return {
    hotels: selectHotels(state.hotels, state.filters)
  };
};
export default connect(
  mapStateToProps,
  undefined
)(HotelList);
