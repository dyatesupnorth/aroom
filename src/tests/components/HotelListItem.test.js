import React from "react";
import { shallow } from "enzyme";
import { HotelListItem } from "../../components/HotelListItem";
import hotels from "../fixtures/hotels";

test("should render HotelListItem with hotel", () => {
  const wrapper = shallow(<HotelListItem {...hotels[0]} />);
  expect(wrapper).toMatchSnapshot();
});


test("should render HotelListItem with hotel and no facilities", () => {
  const wrapper = shallow(<HotelListItem {...hotels[3]} />);
  expect(wrapper).toMatchSnapshot();
});