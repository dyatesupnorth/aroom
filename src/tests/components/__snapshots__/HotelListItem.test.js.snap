// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HotelListItem with hotel 1`] = `
<div
  style={
    Object {
      "border": "1px solid tomato",
    }
  }
>
  <h2>
    hotelone
  </h2>
  <p>
    Star rating: 
    5
  </p>
  <p>
    Facilities:
    <span
      key="car park"
    >
      car park
    </span>
    <span
      key="pool"
    >
      pool
    </span>
  </p>
</div>
`;

exports[`should render HotelListItem with hotel and no facilities 1`] = `
<div
  style={
    Object {
      "border": "1px solid tomato",
    }
  }
>
  <h2 />
  <p>
    Star rating: 
  </p>
  <p>
    No facilities
  </p>
</div>
`;
